File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentRiptide.java
package thedarkcolour.futuremc.enchantment;

import net.minecraft.enchantment.Enchantment;

public class EnchantmentRiptide extends EnchantmentTridentBase {
    public EnchantmentRiptide() {
        super(Rarity.RARE, "riptide");
    }

    @Override
    public int getMinEnchantability(int enchantmentLevel) {
        return 10 + enchantmentLevel * 7;
    }

    @Override
    public int getMaxEnchantability(int enchantmentLevel) {
        return 50;
    }

    @Override
    public int getMaxLevel() {
        return 3;
    }

    @Override
    protected boolean canApplyTogether(Enchantment ench) {
        return super.canApplyTogether(ench) && ench != Enchantments.LEALTAD && ench != Enchantments.CONDUCTIVIDAD;
    }
}

File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentRiptide.kt
package thedarkcolour.futuremc.enchantment

import net.minecraft.enchantment.Enchantment
import thedarkcolour.futuremc.registry.FEnchantments

class EnchantmentRiptide : EnchantmentTridentBase(Rarity.RARE, "riptide") {
    override fun getMinEnchantability(enchantmentLevel: Int): Int {
        return 10 + enchantmentLevel * 7
    }

    override fun getMaxEnchantability(enchantmentLevel: Int): Int {
        return 50
    }

    override fun getMaxLevel(): Int {
        return 3
    }

    override fun canApplyTogether(ench: Enchantment): Boolean {
        return super.canApplyTogether(ench) && ench !== FEnchantments.LEALTAD && ench !== FEnchantments.CONDUCTIVIDAD
    }
}